snapcraft/cli/remote.py

```diff
@@ -0,0 +1,252 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import os
+import uuid
+
+from snapcraft.internal.remote_build import (
+    Worktree,
+    LaunchpadClient,
+    Repo,
+    InfoFile,
+    errors,
+)
+from typing import List, Tuple
+from xdg import BaseDirectory
+from . import echo
+from ._options import get_project
+
+_SUPPORTED_ARCHS = ["amd64", "arm64", "armhf", "i386", "ppc64el", "s390x"]
+
+
+@click.group()
+def remotecli():
+    """Remote build commands"""
+    pass
+
+
+@remotecli.command("remote-build")
+@click.option(
+    "--recover",
+    metavar="<build-number>",
+    type=int,
+    nargs=1,
+    required=False,
+    help="Recover interrupted remote build.",
+)
+@click.option(
+    "--status",
+    metavar="<build-number>",
+    type=int,
+    nargs=1,
+    required=False,
+    help="Display remote build status.",
+)
+@click.option(
+    "--arch",
+    metavar="<arch-list>",
+    type=str,
+    nargs=1,
+    required=False,
+    help="Set architectures to build.",
```

Comment:
the list of valid arches could be passed here as a param to click.option

Reply:
Unfortunately `type=click.Choice()` doesn't seem to fit this situation without changing the parameter format. We can either leave it without the choice list, or change the list separator to space instead of comma. Comma-separated seems more natural to me, other members of a space-separated list feel like they're other arguments:
```
$ snapcraft remote-build --arch=i386,amd64,armhf --git

$ snapcraft remote-build --arch=i386 amd64 armhf --git
```